experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -0,0 +1,493 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as SchemaAware from "../schema-aware";
+import { FieldKey, TreeSchemaIdentifier, TreeValue } from "../../core";
+import { Assume, RestrictiveReadonlyRecord, _InlineTrick } from "../../util";
+import { LocalNodeKey } from "../node-key";
+import {
+	FieldSchema,
+	InternalTypedSchemaTypes,
+	TreeSchema,
+	AllowedTypes,
+	FieldNodeSchema,
+	LeafSchema,
+	MapSchema,
+	StructSchema,
+} from "../typed-schema";
+import { EditableTreeEvents } from "../untypedTree";
+import { FieldKindTypes, FieldKinds } from "../default-field-kinds";
+import { TreeStatus } from "../editable-tree";
+import { TreeContext } from "./context";
+
+/**
+ * Part of a tree.
+ * Iterates over children.
+ *
+ * @privateRemarks
+ * This exists mainly as a place to share common members between nodes and fields.
+ * It is not expected to be useful or common to write code which handles this type directly.
+ * If this assumption turns out to be false, and generically processing `UntypedEntity`s is useful,
+ * then this interface should probably be extended with some down casting functionality (like `is`).
+ *
+ * TODO:
```

Comment:
Intended to be done in this PR? Or in the future? (Feel free to close)

Reply:
very much for later.